src/lib/censorDeath.js

```diff
@@ -0,0 +1,85 @@
+const censorDeath = (death, roles) => {
+  const r = {};
+  if (death.nextRegistration) {
+    r.nextRegistration = censorDeath(death.nextRegistration, roles)
+  }
+  else if (death.previousRegistration) {
```

Comment:
`else if` creates the same problem as before.

Sometimes you have both `nextRegistration` and `previousRegistration`, you need to be able to deal with that.